ocs_ci/ocs/platform_nodes.py

```diff
@@ -1642,3 +1642,95 @@ def read_default_config(self, default_config_path):
             default_config_dict = yaml.safe_load(f)
 
         return default_config_dict
+
+
+class AZURENodes(NodesBase):
+    """
+    Azure Nodes class
+    """
+    def __init__(self):
+        super(AZURENodes, self).__init__()
+        self.subscription_id = config.ENV_DATA.get("azure_subscription_id")
+        self.client_id = config.ENV_DATA['azure_client_id']
+        self.secret = config.ENV_DATA['azure_secret']
+        self.tenant = config.ENV_DATA['azure_tenant']
+        self.resource_group = config.ENV_DATA['azure_resource_group']
```

Comment:
Btw would be better to not store these secrets in this class when all you do with it later is to pass it to instance of `azure_utils.AZURE` class in `self.azure`? In `AWSNodes` I also see only `self.aws` and `self.az` only.

Reply:
Since AWS pattern uses boto3 module, by default it extracts the credentials from `~/.aws/credentials`, so explicitly passing creds to the class is not required.
For AZURE pattern, have followed the approach as in `VMWareNodes` (extracting cred from default_config.yaml and passing it to the class)

Reply:
Ok, I will see later if extracting azure credentials from `~/.azure/osServicePrincipal.json` in a similar way works or makes sense here.